smartsim/_core/launcher/dragon/dragonSockets.py

```diff
@@ -115,20 +117,28 @@ def get_authenticator(
     :type context: zmq.Context
     :returns: the activated `Authenticator`
     :rtype: zmq.auth.thread.ThreadAuthenticator"""
+    # pylint: disable-next=global-statement
+    global AUTHENTICATOR
+
+    if AUTHENTICATOR is not None:
+        if AUTHENTICATOR.is_alive():
+            return AUTHENTICATOR
+        del AUTHENTICATOR
+
     config = get_config()
 
     key_manager = KeyManager(config, as_client=True)
     server_keys, client_keys = key_manager.get_keys()
     logger.debug(f"Applying keys to authenticator: {server_keys}, {client_keys}")
 
-    authenticator = zmq.auth.thread.ThreadAuthenticator(context)
+    AUTHENTICATOR = zmq.auth.thread.ThreadAuthenticator(context)
```

Comment:
Do we want to get rid of `get_authenticator` if we're creating these directly? Do we want to use `get_authenticator` to ensure security is applied uniformly?